### DISCUSSION TITLE
Question: Where is String.Dump() extension method?

### DISCUSSION THREAD

**Discussion by bancroftway** [JoshClose/CsvHelper#1613] [Q&A]
I am trying to use CsvHelper to write a datatable to csv, and found some sample code [here](https://github.com/JoshClose/CsvHelper/issues/1399).

I have installed the latest version (15.0.10) of CsvHelper from nuget. 

However, I cannot locate the Dump() extension method on string class. Where is this declared, and what namespace do I need to import? Curiously, Visual Studio intellisense cannot suggest the namespace to import, so maybe something got changed? Thanks

**Comment by philjones** [comment]
I'm not the original author of that example but I've encountered `Dump()` when using [LINQPad ](https://www.linqpad.net/#dump).

It's a pretty awesome feature of the code scratchpad, so it could be the author of the example was using that.

**Comment by bancroftway** [comment]
> I'm not the original author of that example but I've encountered `Dump()` when using [LINQPad ](https://www.linqpad.net/#dump).
> 
> It's a pretty awesome feature of the code scratchpad, so it could be the author of the example was using that.

@philjones The Dump() method is used many times in various CsvHelper examples, not just the quoted sample. In fact IEnumerables have a Dump() extension method as shown [here](https://github.com/JoshClose/CsvHelper/issues/1451#issuecomment-582095515). So I am guessing, I am just missing a namespace include, but just not sure what it is.

**Comment by AltruCoder** [comment]
@bancroftway Josh uses LINQPad when he is creating those examples.  It confused me too trying to figure out where that extension method was, but he confirmed that was what he was using.

**Comment by JoshClose** [comment]
Yes, it's an extension method included in LINQPad. It prints object graphs in a graphical manner. I use LINQPad for all my examples because it's so fast to get an example up and running. I highly recommend it.
